pkgs/development/compilers/llvm/3.8/llvm.nix

```diff
@@ -18,6 +18,18 @@
 
 let
   src = fetch "llvm" "1ybmnid4pw2hxn12ax5qa5kl1ldfns0njg8533y3mzslvd5cx0kf";
+
+  llvmTarget = platform:
+    if platform.parsed.cpu.family == "x86" then
+      "X86"
+    else if platform.parsed.cpu.name == "aarch64" then
+      "AArch64"
+    else if platform.parsed.cpu.family == "arm" then
+      "ARM"
+    else if platform.parsed.cpu.family == "mips" then
+      "Mips"
+    else
+      throw "Unsupported system";
```

Comment:
Perhaps a "switch"-style instead here?

```nix
{
  mips = "Mips";
  x86 = "X86";
   ...
}."${platform.parsed.cpu.family}" or throw "Unsupported system"
```

(or thereabouts)

Reply:
Shoudn't LLVM_TARGETS_TO_BUILD default to all? Not sure what the rust error is though:

https://llvm.org/docs/CMake.html#llvm-specific-variables

Reply:
If it does then yes we can remove any mention of target and share llvms. Just compiler-rt would be rebuilt.

Reply:
Should we be concerned about output size/build times with this? LLVM_TARGETS_TO_BUILD is probably being overriden from the default right now when TARGET_TRIPLE is set. I kind of like that GCC's target-specific stuff doesn't give you anything for free (you have to set depsBuildBuild, nativeBuildInputs where appropriate). We might also still need DEFAULT_TARGET_TRIPLE.

Reply:
I did not do that because libllvm is also used for jit compiling and therefore a dependency of some applications. Do you think building all backends would have many users? Most of the time I think one picks one host platform to compile for and this is also how our cross-buildsystem is designed. So it won't be useful for many people to have all backends.

Reply:
@dtzWill the switch case style does not work here because of I have to use `platform.parsed.cpu.name` instead of family for aarch64.